exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/batch/BatchFileService.java

```diff
@@ -164,4 +170,14 @@ public BatchId getLatestBatchIdV2(BatchIntervals intervals) {
             return BatchFileFactory.createBatchFile(signatureConfig, signingKey, metadata, keys);
         }
     }
+
+    private int generateDemoId(int keyCount, int base) {
+        if (keyCount < base) {
+            return keyCount;
+        } else if (keyCount % base == 0) {
+            return base - 1;
```

Comment:
So it gives the 9999 id twice? Why? I assume the point is to avoid overflowing the integer with an extra digit, but I don't see how repeating the last number is better than starting again from 0.

Reply:
it uses it everytime modulo is zero, which is not nice. In practice this affects only our test environments where total keys count is millions.